packages/rule-schema-to-typescript-types/src/index.ts

```diff
@@ -78,10 +83,16 @@ export function compile(
   const optionsType = isArraySchema
     ? `type Options = [${typeStrings.join(',')}]`
     : `type Options = ${typeStrings[0]}`;
-  return prettierFormat(
-    [...refTypes, optionsType].join('\n\n'),
-    prettierConfig,
+
+  const codeWithUnsimplifiedOptions = [...refTypes, optionsType].join('\n\n');
+  const simplifiedOptionsType = useTsToSimplifyOptionsType(
+    codeWithUnsimplifiedOptions,
+  );
+  const codeWithSimplifiedOptions = [...refTypes, simplifiedOptionsType].join(
+    '\n\n',
```

Comment:
can we correct order of items, to reproduce how this was generated before?

in main/v6 we print defaults and options in same order

main:
![image](https://user-images.githubusercontent.com/625469/232794617-4d063edc-54c1-4d6b-a18b-21d39a88da60.png)

this branch:
![image](https://user-images.githubusercontent.com/625469/232794499-e7e6eba1-8ca2-4cdd-a246-67888597a56f.png)

Reply:
yeah I can update the defaults printer so that it sorts alphabetically as well!

Reply:
> yeah I can update the defaults printer so that it sorts alphabetically as well!

can we use `natural-compare` instead of `localeCompare` for this?

that should sort it as `'array' | 'array-simple' | 'generic'` instead of `'array-simple' | 'array' | 'generic'`

Reply:
After playing for a bit - natural compare isn't a good fit for code, IMO.

For example natural compare will sort tuples in a nearly reverse order to what we'd want:
```ts
[
  "['always']",
  "['never',{ignoreForLoopInit?: boolean;},'option']",
  "['never',{ignoreForLoopInit?: boolean;},]",
  "['never']",
  '[]'
]

// we really want this order:
[
  '[]',
  "['always']",
  "['never']",
  "['never',{ignoreForLoopInit?: boolean;},]",
  "['never',{ignoreForLoopInit?: boolean;},'option']",
]
```

Unfortunately it ignores whitespace amongst non-letters, so we can't control this without doing some really janky stuff.
I'll stick with localeCompare instead